programs/mmm/src/util.rs

```diff
@@ -547,3 +566,109 @@ pub fn assert_valid_fees_bp(maker_fee_bp: i16, taker_fee_bp: i16) -> Result<()>
 
     Ok(())
 }
+
+pub fn check_buyside_sol_escrow_account(
+    pda: &Pubkey,
+    pool_key: &Pubkey,
+    buyer: &Pubkey,
+) -> Result<Pubkey> {
+    let (single_pool_pda, _) = Pubkey::find_program_address(
+        &[
+            BUYSIDE_SOL_ESCROW_ACCOUNT_PREFIX.as_bytes(),
+            pool_key.as_ref(),
+        ],
+        &ID,
+    );
+
+    if single_pool_pda == *pda {
```

Comment:
seems like this check is not needed, we only support the shared escrow here

Reply:
refactored